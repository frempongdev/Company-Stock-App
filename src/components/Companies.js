import React, { useState } from 'react';
import CompanyList from './CompanyList';
import '../styles/Companies.css';

const Companies = () => {
  const [searchTerm, setSearchTerm] = useState('');

  return (
    <div className="main">
      <div className="main-top">
        <input type="text" className="search" placeholder="Search..." value={searchTerm} onChange={(e) => setSearchTerm(e.target.value)} />
        <br />
        <h4 className="header">World&rsquo;s Most Valuable Company Stocks</h4>
      </div>
      <div className="company-box">
        <CompanyList searchTerm={searchTerm} />
        <button type="button" className="back-to-top" onClick={() => window.scrollTo({ top: 0, behavior: 'smooth' })}>^Back to top^</button>
      </div>
    </div>
  );
};

export default Companies;
