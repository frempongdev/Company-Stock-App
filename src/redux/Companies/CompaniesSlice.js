import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const url = 'https://financialmodelingprep.com/api/v3/stock-screener?priceMoreThan=150&exchange=NASDAQ,NYSE&apikey=d97b2f08a9a36a89179abfa4fb580330';

const initialState = {
  companies: [],
  isLoading: true,
  error: '',
};

export const fetchCompanies = createAsyncThunk('companies/fetchCompanies', async () => {
  const response = await fetch(url);
  const data = response.json();
  // console.log('nice', data);
  return data;
});

const companiesSlice = createSlice({
  name: 'companies',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchCompanies.pending, (state) => ({ ...state, isLoading: true }))
      .addCase(fetchCompanies.fulfilled, (state, action) => (
        { ...state, isLoading: false, companies: action.payload }))
      .addCase(fetchCompanies.rejected, (state) => ({ ...state, isLoading: false, error: 'company fetch was not successful' }));
  },
});

export default companiesSlice.reducer;
