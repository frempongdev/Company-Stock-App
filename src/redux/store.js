import { configureStore } from '@reduxjs/toolkit';
import companiesReducer from './Companies/CompaniesSlice';

const store = configureStore({
  reducer: {
    companies: companiesReducer,
  },
});

export default store;
