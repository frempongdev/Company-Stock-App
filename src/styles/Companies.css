.main-top {
  background-image: url(../assets/stocks-bg.png);
  background-position: -84% 80%;
  background-size: 68%;
  background-repeat: no-repeat;
  border-bottom: rgb(255, 255, 255) solid;
}

.back-to-top {
  position: fixed;
  bottom: 0;
  right: 2%;
}

.search {
  border: none;
  border-radius: 5px;
  float: right;
  margin-top: 2%;
  padding: 0.5%;
}

.header {
  margin: 6%;
  color: rgb(55, 55, 55);
  font-size: 1.01rem;
}
