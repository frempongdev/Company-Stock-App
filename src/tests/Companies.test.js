import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Companies from '../components/Companies';
import store from '../redux/store';

describe('Companies', () => {
  it('renders Companies', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Companies />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
