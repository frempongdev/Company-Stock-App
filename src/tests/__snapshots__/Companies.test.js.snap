// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies renders Companies 1`] = `
<div
  className="main"
>
  <h5
    className="header"
  >
    Worldâ€™s Most Valuable Company Stocks
  </h5>
  <input
    className="search"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <div
    className="company-box"
  >
    <div
      className="company-wrapper"
    />
    <button
      className="back-to-top"
      onClick={[Function]}
      type="button"
    >
      ^Back to top^
    </button>
  </div>
</div>
`;

exports[`Profile renders Profile 1`] = `
<div>
  <button
    className="back"
    onClick={[Function]}
    type="button"
  >
    &lt;
    back
  </button>
  <div
    className="company-img"
  >
    <img
      alt="companyImg"
    />
  </div>
  <div
    className="company-details"
  >
    <h2
      className="company-name"
    />
    <h6
      className="company-location"
    >
      ,
       
    </h6>
    <p
      className="company-symbol"
    >
      Stock Symbol:
       
    </p>
    <p
      className="stock"
    >
      Stock Price:
       
      $
    </p>
    <p
      className="industry"
    >
      Industry :
       
    </p>
    <h4
      className="details"
    >
      Company Overview
    </h4>
    <p
      className="details"
    />
  </div>
</div>
`;
